packages/compass-connections-navigation/src/base-navigation-item.tsx

```diff
@@ -114,12 +114,13 @@ export const NavigationBaseItem: React.FC<NavigationBaseItemProps> = ({
       <div className={cx('item-wrapper', itemWrapperStyles)} style={style}>
         {isExpandVisible && (
           <ExpandButton
-            onClick={(evt) => {
-              if (isExpandDisabled) return;
-              evt.stopPropagation();
-              onExpand(!isExpanded);
+            onClick={(event) => {
+              // Prevent the click from propagating to the `TreeItem`, triggering the default action
+              event.stopPropagation();
+              toggleExpand();
             }}
             isExpanded={isExpanded}
+            disabled={isExpandDisabled}
```

Comment:
Based on what Betsy's message, I was under the impression we were going to hide the caret altogether, not make it disabled: https://mongodb.slack.com/archives/G2L10JAV7/p1731623253437699?thread_ts=1731413469.751959&cid=G2L10JAV7 
I feel showing it in the UI makes me want to click it. I feel we should either make it not show up at all or make it clickable to connect.

Reply:
@bsradcliffe wanted to keep them (see https://github.com/mongodb-js/compass/pull/6449#issuecomment-2471305136)
> I would like to preserve the chevrons.

Reply:
Cool cool, fine as is then. I'll ping @bsradcliffe to see if making them clickable to connect is an option, or if we want to keep it as is with the disabled.